tests/core/test_datamodules.py

```diff
@@ -399,14 +426,23 @@ def transfer_batch_to_device(self, data, device):
     trainer = Trainer(gpus=1)
     # running .fit() would require us to implement custom data loaders, we mock the model reference instead
     trainer.get_model = MagicMock(return_value=model)
+    if is_overridden('on_before_batch_transfer', dm):
+        model.on_before_batch_transfer = dm.on_before_batch_transfer
     if is_overridden('transfer_batch_to_device', dm):
         model.transfer_batch_to_device = dm.transfer_batch_to_device
+    if is_overridden('on_after_batch_transfer', dm):
+        model.on_after_batch_transfer = dm.on_after_batch_transfer
```

Comment:
Why do we have the `is_overriden` checks? They are overriden

Reply:
all these hooks are a part of `DataHooks` and both `LightningModule` and `LightningDataModule` inherit it. So I guess it's some kind of priority logic followed here. If the method is overridden in both then the one overridden in DataModule is taken here else it will continue with the one defined in LightningModule.

Reply:
I get that, but for this test in particular, the if will always resolve to True, so why have it?

Reply:
okay yeah, I didn't see the filename here. I thought it's from the other file.